addons/binding/org.openhab.binding.modbus.sunspec/src/main/java/org/openhab/binding/modbus/sunspec/internal/discovery/SunspecDiscoveryParticipant.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.modbus.sunspec.internal.discovery;
+
+import static org.openhab.binding.modbus.sunspec.internal.SunSpecBindingConstants.SUPPORTED_THING_TYPES_UIDS;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.openhab.binding.modbus.discovery.ModbusDiscoveryListener;
+import org.openhab.binding.modbus.discovery.ModbusDiscoveryParticipant;
+import org.openhab.binding.modbus.handler.EndpointNotInitializedException;
+import org.openhab.binding.modbus.handler.ModbusEndpointThingHandler;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Discovery service for sunspec
+ *
+ * @author Nagy Attila Gabor - initial contribution
+ *
+ */
+@Component(immediate = true)
+public class SunspecDiscoveryParticipant implements ModbusDiscoveryParticipant {
+
+    private final Logger logger = LoggerFactory.getLogger(SunspecDiscoveryParticipant.class);
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypeUIDs() {
+        return new HashSet<ThingTypeUID>(SUPPORTED_THING_TYPES_UIDS.values());
+    }
+
+    @Override
+    public void startDiscovery(ModbusEndpointThingHandler handler, ModbusDiscoveryListener listener) {
+        logger.trace("Starting sunspec discovery");
+        try {
+            new SunspecDiscoveryProcess(handler, listener).detectModel();
+        } catch (EndpointNotInitializedException ex) {
+            logger.debug("Could not start discovery process");
```

Comment:
Raise to "warn". Its an actual error that should be highlighted in the logs don't you think.
And also: Add the exception as last argument. That adds a stacktrace and the exception message to the log.